src/v/cluster/tests/partition_balancer_planner_fixture.h

```diff
@@ -0,0 +1,169 @@
+          .node_availability_timeout_sec
+          = config::mock_binding<std::chrono::seconds>(1min)})
+      , last_node_idx(0) {
+        table.start().get0();
+        members.start_single().get0();
+        allocator
+          .start_single(
+            std::ref(members),
+            config::mock_binding<std::optional<size_t>>(std::nullopt),
+            config::mock_binding<std::optional<int32_t>>(std::nullopt),
+            config::mock_binding<size_t>(16_KiB),
+            config::mock_binding<bool>(false))
+          .get0();
+    }
+
+    ~partition_balancer_fixture() {
+        table.stop().get0();
+        allocator.stop().get0();
+        members.stop().get0();
+    }
+
+    cluster::topic_configuration_assignment make_tp_configuration(
+      const ss::sstring& topic, int partitions, int16_t replication_factor) {
+        cluster::topic_configuration cfg(
+          test_ns, model::topic(topic), partitions, replication_factor);
+
+        cluster::allocation_request req;
+        req.partitions.reserve(partitions);
+        for (auto p = 0; p < partitions; ++p) {
+            req.partitions.emplace_back(
+              model::partition_id(p), replication_factor);
+        }
+
+        auto pas = allocator.local()
+                     .allocate(std::move(req))
+                     .value()
+                     .get_assignments();
+
+        return cluster::topic_configuration_assignment(cfg, std::move(pas));
+    }
+
+    cluster::create_topic_cmd make_create_topic_cmd(
+      const ss::sstring& name, int partitions, int replication_factor) {
+        return cluster::create_topic_cmd(
+          make_tp_ns(name),
+          make_tp_configuration(name, partitions, replication_factor));
+    }
+
+    model::topic_namespace make_tp_ns(const ss::sstring& tp) {
+        return model::topic_namespace(test_ns, model::topic(tp));
+    }
+
+    void create_topic(
+      const ss::sstring& name, int partitions, int replication_factor) {
+        auto cmd = make_create_topic_cmd(name, partitions, replication_factor);
+        auto res = table.local().apply(std::move(cmd), model::offset(0)).get0();
+        BOOST_REQUIRE_EQUAL(res, cluster::errc::success);
+    }
+
+    void allocator_register_nodes(size_t nodes_amount) {
+        for (size_t i = 0; i < nodes_amount; ++i) {
+            allocator.local().register_node(
+              create_allocation_node(model::node_id(last_node_idx), 4));
+            last_node_idx++;
+        }
+    }
+
+    std::vector<raft::follower_metrics> create_follower_metrics() {
+        std::vector<raft::follower_metrics> metrics;
+        metrics.reserve(last_node_idx);
+        for (size_t i = 0; i < last_node_idx; ++i) {
+            metrics.push_back(raft::follower_metrics{
+              .id = model::node_id(i),
+              .last_heartbeat = raft::clock_type::now()});
+        }
+        return metrics;
+    }
+
+    cluster::cluster_health_report create_health_report() {
```

Comment:
Not necessary to implement this right now, but it would be great to be able to specify arbitrary disk sizes, partition sizes and partition placement for tests and to have a "test descriptor" structure that allows easily specifying and viewing it.